narwhals/_arrow/dataframe.py

```diff
@@ -602,31 +602,35 @@ def is_duplicated(self: Self) -> ArrowSeries:
 
         from narwhals._arrow.series import ArrowSeries
 
-        df = self._native_frame
+        index_token = generate_temporary_column_name(n_bytes=8, columns=self.columns)
+        df = self.with_row_index(index_token)._native_frame
 
         columns = self.columns
         col_token = generate_temporary_column_name(n_bytes=8, columns=columns)
```

Comment:
Very unlikely, but `columns` here does not include `index_token` (for the comment below)